### PR TITLE
Fix metrics not shutting down if there are open connections

### PR DESCRIPTION
**Pull request by tmpolaczyk**
Fix prometheus metrics not shutting down if there are open connections. I fixed the same issue in the past but it broke again after a dependecy upgrade.

See also:

https://github.com/paritytech/polkadot-sdk/pull/1637

### CODE REVIEW

**Comment by bkchr** [comment]
> I fixed the same issue in the past but it broke again after a dependecy upgrade.

Do we want to prevent this again from happening? By introducing a test? :D

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
@tmpolaczyk also please provide some prdoc.

**Review by lexnv — APPROVED** [review]

**Comment by tmpolaczyk** [comment]
@bkchr not sure if this is easy to test. I have a zombienet test in our project that will fail if this breaks, but it takes 5 minutes to run.

For a unit test we would need to:
* Start metrics on port N
* Create tcp client and connect to port N
* Stop metrics
* Try to start something else on port N, will fail because the port is still being used

**Review by skunert — APPROVED** [review]

File: prdoc/pr_6220.prdoc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+title: Improve `CheckMetadataHash` transaction extension weight and logic
```
Comment (pepoviola) [inline prdoc/pr_6220.prdoc:1]: `title` needs to be updated, to reference the actual fix.

Comment (tmpolaczyk) [reply prdoc/pr_6220.prdoc]: :see_no_evil: thanks

**Comment by girazoki** [comment]
@bkchr is there anything else to do before merging this PR?

**Comment by bkchr** [comment]
No! Sorry for forgetting! :sweat_smile:

**Pull request merged**